src/python/dxpy/scripts/dx.py

```diff
@@ -3785,6 +3785,158 @@ def publish(args):
+            else: 
+                eprint("     Unexpected error: {}".format(format_exception(e)))
+            
+            err_exit("Failed request: {}. {}".format(request_input, format_exception(e)), code=3)
+        return res              
+
+    def get_valid_archival_input(args, target_files, target_folder, target_project):
+        request_input = {}
+        if not target_project:
+            err_exit('No target project has been set. Please check the input or check your permission to the given project.', code=3)
+        if target_files and target_folder:
+            err_exit('Expecting either a single folder or a list files for each API request', code=3)
+        elif target_files: 
+            target_files = list(target_files)
+            request_input = {"files": target_files}
+        elif target_folder:
+            request_input = {"folder": target_folder, "recurse":args.recurse}
+        else:
+            err_exit("No input file/folder is found in project {}".format(target_project), 
+                            code=3)
+        return request_input
+
+    def get_archival_paths(args):
+        target_project = None
+        target_folder = None
+        target_files = set()
+        
+        paths = [split_unescaped(':', path, include_empty_strings=True) for path in args.path]
+        possible_projects = set()
+        possible_objects = set()
+        for p in paths:
+            if len(p)>2: 
+                err_exit("Path '{}' is invalid. Please check the inputs or check --help for example inputs.".format(p), code=3)
+            elif len(p) == 2:
+                possible_projects.add(p[0])
+            possible_objects.add(p[-1])
+
+        # get project ID
+        for proj in possible_projects:
+            # is project ID
+            if is_container_id(proj):
+                pass
+            # is "": use current project
+            elif proj == '':
+                if not dxpy.PROJECT_CONTEXT_ID:
+                    err_exit("Cannot find current project. Please check the environment.", code=3)
+                proj = dxpy.PROJECT_CONTEXT_ID
+            # name is given
+            else:
+                try:
+                    project_results = list(dxpy.find_projects(name=proj, describe=True))
+                    if project_results:
+                        choice = pick(["{} ({})".format(result['describe']['name'], result['id']) for result in project_results], allow_mult=False)
+                        proj = project_results[choice]['id']
+                except:
+                    err_exit("Cannot find project with name {}".format(proj), code=3)
+
+            if target_project and proj!= target_project:
+                err_exit("All paths must refer to files/folder in a single project, but two project ids: '{}' and '{}' are given. ".format(
+                                target_project, proj), code=3)
+            else:
+                target_project = proj
```

Comment:
@jtratner 
- multiple projects would raise an error
- we keep extending the file array from the inputs until there is one input seen as a folder path
- later there is a validation check: if both file arrays and a folder path are non empty, it would raise an error